config.py

```diff
@@ -47,6 +64,15 @@ class vedaAppSettings(BaseSettings):
         description="Custom domain name, i.e. veda-backend.xyz",
     )
 
+    oidc_provider_arn: Optional[AwsOidcArn] = Field(
+        description="ARN of AWS OIDC provider used for authentication"
+    )
+
+    oidc_repo_id: str = Field(
+        "NASA-IMPACT/veda-backend",
+        description="ID of AWS ECR repository used for OIDC provider",
+    )
+
```

Comment:
I don't know if we should do OIDC stuff via CDK, the role policy creation has to be manual anyway. In MCP, OIDC provider can't be updated, and we have to make a request for the role creation so it's easier to share it across multiple deployments.

Reply:
I'm not as familiar with what it takes to get OIDC to work in MCP. I'm inclined to leave it in for now, since it is optional and will help with creating actions for our dev environments, but we can look at refactoring this functionality to use a shared role.